lib/game_play.rb

```diff
@@ -25,16 +24,16 @@ def mark_board(board, player_mark, spot)
          board[spot - 1] = player_mark
     end
 
-    def select_spot(board)
-        @ui.display_message(@messages.lookup(:get_spot))
-        spot = @ui.get_spot_input
-        is_valid = @validate.valid_free_spot?(spot, board)
-        if !is_valid
-            @ui.display_message(@messages.lookup(:invalid_input)) 
-            return select_spot(board)
-        else
-            return spot
-        end
+    def select_spot(board, player_mark)
+      @ui.display_message(@messages.lookup(:"#{player_mark}_get_spot"))
+      spot = @ui.get_spot_input
+      is_valid = @validate.valid_free_spot?(spot, board)
+      if !is_valid
```

Comment:
just wanted to comment that -  while you could inline this if.  I personally like that I can see what it means without having to decipher the code.